logstash-core/lib/logstash/util/substitution_variables.rb

```diff
@@ -37,8 +37,10 @@ def deep_replace(value)
       end
     else
       if value.is_a?(Array)
-        value.each_index do | valueArrayIndex|
-          value[valueArrayIndex] = deep_replace(value[valueArrayIndex])
+        valueArrayIndex = 0
+        value.each do |single_value|
+          value[valueArrayIndex] = deep_replace(single_value)
+          valueArrayIndex += 1
```

Comment:
Styleguide nitpick: use snake_case for variables in ruby code

```suggestion
        value_array_index = 0
        value.each do |single_value|
          value[value_array_index] = deep_replace(single_value)
          value_array_index += 1
```